syntax = "proto3";

option csharp_namespace = "AcademyCloud.ResourceManagement.Protos.Identity";

package identity;

service Identity {
  rpc AddUser (AddUserRequest) returns (AddUserResponse);
  rpc RemoveUserFromProject (RemoveUserFromProjectRequest) returns (RemoveUserFromProjectResponse);
  rpc DeleteProject (DeleteProjectRequest) returns (DeleteProjectResponse);
  rpc DeleteUser (DeleteUserRequest) returns (DeleteUserResponse);
}

message AddUserRequest {
  string userProjectAssignmentId = 1;
  string userId = 2;
  string projectId = 3;
}

message AddUserResponse {
  
}

message DeleteUserRequest {
  string userId = 1;
}

message DeleteUserResponse {
  
}

message RemoveUserFromProjectRequest {
  string userId = 1;
  string projectId = 2;
}

message RemoveUserFromProjectResponse {
  
}

message DeleteProjectRequest {
  string projectId = 1;
}

message DeleteProjectResponse {
  
}

